src/FamilyHubs.ServiceDirectory.Core/Commands/Organisations/DeleteOrganisation/DeleteOrganisationCommand.cs

```diff
@@ -0,0 +1,80 @@
+﻿using AutoMapper;
+using FamilyHubs.ServiceDirectory.Core.Exceptions;
+using FamilyHubs.ServiceDirectory.Data.Entities;
+using FamilyHubs.ServiceDirectory.Data.Repository;
+using FamilyHubs.SharedKernel.Identity;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace FamilyHubs.ServiceDirectory.Core.Commands.Organisations.DeleteOrganisation;
+
+public class DeleteOrganisationCommand : IRequest<bool>
+{
+    public DeleteOrganisationCommand(long id)
+    {
+        Id = id;
+    }
+
+    public long Id { get; }
+}
+
+public class DeleteOrganisationCommandHandler : IRequestHandler<DeleteOrganisationCommand, bool>
+{
+    private readonly ApplicationDbContext _context;    
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly ILogger<DeleteOrganisationCommandHandler> _logger;
+
+    public DeleteOrganisationCommandHandler(ApplicationDbContext context, IHttpContextAccessor httpContextAccessor,
+        ILogger<DeleteOrganisationCommandHandler> logger)
+    {
+        _context = context;        
+        _httpContextAccessor = httpContextAccessor;
+        _logger = logger;
+    }
+
+    public async Task<bool> Handle(DeleteOrganisationCommand request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var entity = _context.Organisations.SingleOrDefault(o => o.Id == request.Id);
+            if (entity is null)
+                throw new NotFoundException("Organisation Not Found");
+
+            ThrowIfForbidden(request, entity);
+
+            _context.Organisations.Remove(entity);
```

Comment:
Does this actually cascade delete?

Reply:
Hi Martin, Good point. Yes it does I have tested it.